back/app/services/timeline_service.rb

```diff
@@ -105,6 +105,11 @@ def phase_number(phase)
   end
 
   def previous_phase(phase)
-    Phase.where('project_id = ? AND start_at < ?', phase.project_id, phase.start_at).order(start_at: :desc).take
+    Phase
+      .where.not(id: phase.id)
```

Comment:
This is the fix. The same phase would be returned as the previous phase, because the start date was increased but this new start date was not persisted in the database yet.